src/gateway/app.py

```diff
@@ -1,81 +1,122 @@
-import requests, time
-
-from flask import Flask, request, make_response 
+import requests
+import os
+from flask import Flask, request, jsonify, make_response
 from requests.exceptions import ConnectionError, HTTPError
-from pymongo.errors import ServerSelectionTimeoutError
-
-from werkzeug.exceptions import NotFound
-
-ALLOWED_USER_OPS = ['register', 'login', 'admin/register', 'admin/login', 'add-balance', 'profile']
-ALLOWED_GATCHA_OPS = ['roll', 'get-gatcha-id', 'get-all-gatcha', 'add-gatcha', 'delete-gatcha']
-ALLOWED_MARKET_OPS = ['create-auction', 'bid', 'get-auction', 'add-auction', 'delete-auction']
-
-#CHANGE URLS TO MATCH THE NAMES AND PORTS OF THE SERVICES IN THE DOCKER-COMPOSE FILE
-GATCHA_URL = 'http://gatcha:5000'
-USER_URL = 'http://user:5000'
-MARKET_URL = 'http://market:5000'
-DB_MANAGER_URL = 'http://db-manager:5000'
-
+ALLOWED_USER_OPS = {
+    'register': 'POST',
+    'login': 'POST',
+    'profile': 'GET',
+    'add-balance': 'POST',
+    'get_user_from_name': 'GET'
+}
 
-ids = {} #CAREFUL, THIS IS NOT FOR MULTIUSER AND MULTITHREADING, JUST FOR DEMO PURPOSES
+# URLs for services based on environment variables
+GATCHA_URL = os.getenv('GATCHA_URL', 'http://gatcha:5000')
+USER_URL = os.getenv('USER_URL', 'http://user:5000')
+MARKET_URL = os.getenv('MARKET_URL', 'http://market:5000')
+DB_MANAGER_URL = os.getenv('DBM_URL', 'http://db-manager:5000')
 
-app = Flask(__name__, instance_relative_config=True)
+app = Flask(__name__)
 
-def create_app():
-    return app
-
-def service_request(url, data=None):
+def service_request(url, data):
     try:
         if data:
             response = requests.post(url, json=data)
         else:
             response = requests.get(url)
         response.raise_for_status()
         return response.json()
-    except ConnectionError:
-        return make_response('Gatcha service is down\n', 500)
-    except HTTPError:
-        return make_response(response.content, response.status_code)
+    except Exception as e:
+        print(f"Error making request to {url}: {str(e)} with data{str(data)}")  # Log error
+        return None
+
+# Endpoint per la registrazione degli utenti
+@app.route('/user/register/<username>/<password>')
```

Comment:
Forse è meglio passare password e username in un json, altrimenti questi dati sono visibili nell'URL